blacklight-cornell/app/controllers/catalog_controller.rb

```diff
@@ -17,6 +17,31 @@ class CatalogController < ApplicationController
     #prepend_before_action :set_return_path
   end
 
+
+# ==============================================================================
+# Determines if a new search session should be saved to search history
+# Advanced: true if any q_row has text, or a full date range is set, or
+#           language facet is present, or any inclusive facets are present.
+# Basic:    true if query (q) is present or any facet (f) is present.
+# ------------------------------------------------------------------------------
+  def start_new_search_session?
+    # Advanced search
+    if params[:search_field] == "advanced"
+      inclusive_present = params.dig(:f_inclusive).present?
+      language_present  = params.dig(:f, :language_facet).present?
+      dates_present     = params.dig(:range, :pub_date_facet, :begin).present? && params.dig(:range, :pub_date_facet, :end).present?
+      query_rows, index = params[:q_row], action_name == "index"
+      queries_present   = query_rows.present? && query_rows.any? { |q| q.present? }
+
+      start_new_search_history_record = index && (queries_present || dates_present || language_present || inclusive_present)
+    else
+      # Basic search
+      query, facet, index = params["q"], params["f"], action_name == "index"
+      start_new_search_history_record = index && (query.present? || facet.present? )
+    end
+    start_new_search_history_record # Return true/false
```

Comment:
Someday we could probably just use `search_state.has_constraints?`, instead of duplicating logic. But for now, could it be cleaner to do something like:

`action_name == 'index' && !(search_state.query_param.blank? && (search_state.advanced_query_param.blank? || search_state.advanced_query_param.try(:all?, &:blank?)) && search_state.filters.blank?)`

(advanced query_param would need to be made a public method in BlacklightCornell::SearchState for that to work.)

That way we're hardcoding params a bit less and reusing existing blacklight methods.